src/workerd/server/server.c++

```diff
@@ -760,9 +762,13 @@ class Server::ExternalHttpService final: public Service, private kj::TaskSet::Er
       if (parent.rewriter->needsRewriteResponse()) {
         auto rewrite = headers.cloneShallow();
         parent.rewriter->rewriteResponse(rewrite);
-        return response.send(statusCode, statusText, rewrite, expectedBodySize);
+        auto res = response.send(statusCode, statusText, rewrite, expectedBodySize);
+        wrappedResponse = kj::none;
+        return res;
```

Comment:
This doesn't feel related. Does it fix a bug?

Reply:
Yeah, this service can only be used for one request without this, a true shame.

Reply:
Though we have a problem that the API wants to use this service for concurrent requests which seems to not work so maybe we'll need another approach anyway, we'll see.